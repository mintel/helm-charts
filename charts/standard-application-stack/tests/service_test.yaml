suite: Test Services
templates:
  - service.yaml
release:
  namespace: test-namespace
tests:
  - it: Check name, namespace and labels
    set:
      global.name: test-app
      component: exporter
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: Service
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: test-app
      - equal:
          path: metadata.labels["app.kubernetes.io/name"]
          value: test-app
      - equal:
          path: metadata.labels["name"]
          value: test-app

  - it: Check default port set
    set:
      global.name: test-app
      component: exporter
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: Service
      - equal:
          path: spec.ports[0].port
          value: 8000

  - it: Check port can be overridden
    set:
      global.name: test-app
      component: exporter
      port: 1234
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: Service
      - equal:
          path: spec.ports[0].port
          value: 1234
      - equal:
          path: spec.ports[0].targetPort
          value: 1234

  - it: Check port and targetPort can be set from Service
    set:
      global.name: test-app
      component: exporter
      port: 1234
      service:
        port: 22
        targetPort: 2222
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: Service
      - equal:
          path: spec.ports[0].port
          value: 22
      - equal:
          path: spec.ports[0].targetPort
          value: 2222

  - it: Check extraPorts
    set:
      global.name: test-app
      component: exporter
      extraPorts:
        - name: p1
          containerPort: 8081
        - name: p2
          containerPort: 8082
        - name: p3
          containerPort: 8083
          containerName: custom
        - name: p4
          containerPort: 8084
          shouldIgnoreThis: value
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: Service
      - equal:
          path: spec.ports[1].name
          value: main-p1
      - equal:
          path: spec.ports[1].port
          value: 8081
      - equal:
          path: spec.ports[2].name
          value: main-p2
      - equal:
          path: spec.ports[2].port
          value: 8082
      - equal:
          path: spec.ports[3].name
          value: custom-p3
      - equal:
          path: spec.ports[3].port
          value: 8083
      - equal:
          path: spec.ports[4].port
          value: 8084

  - it: Check ALB annotation can be set when enabled
    set:
      global.name: test-app
      global.owner: my-team
      global.application: my-app
      global.clusterEnv: qa
      ingress.enabled: true
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: Service
      - equal:
          path: metadata.annotations["alb.ingress.kubernetes.io/tags"]
          value: Owner=my-team,Application=my-app

  - it: Check ALB annotation not set when Owner and Application missing
    set:
      global.name: test-app
      ingress.enabled: true
      # Add annotations to allow for isNull check
      service.annotations:
        k: v
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: Service
      - isNull:
          path: metadata.annotations["alb.ingress.kubernetes.io/tags"]

  - it: Check ALB annotation can set Application without Owner
    set:
      global.name: test-app
      ingress.enabled: true
      global.application: my-app
      # Add annotations to allow for isNull check
      service.annotations:
        k: v
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: Service
      - equal:
          path: metadata.annotations["alb.ingress.kubernetes.io/tags"]
          value: Application=my-app

  - it: Check ALB annotation can set Owner without Application
    set:
      global.name: test-app
      ingress.enabled: true
      global.owner: my-team
      # Add annotations to allow for isNull check
      service.annotations:
        k: v
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: Service
      - equal:
          path: metadata.annotations["alb.ingress.kubernetes.io/tags"]
          value: Owner=my-team
