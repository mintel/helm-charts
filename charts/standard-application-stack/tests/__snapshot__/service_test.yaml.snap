Check ALB annotation can be set when enabled:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        alb.ingress.kubernetes.io/tags: Owner=my-team
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: qa
        app.mintel.com/owner: my-team
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      ports:
        - name: main-http
          port: 8000
          targetPort: 8000
      selector:
        app.kubernetes.io/component: app
        app.kubernetes.io/name: test-app
      type: ClusterIP
Check ALB annotation not set when ALB Ingress is disabled:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        k: v
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: local
        app.mintel.com/owner: my-team
        app.mintel.com/region: local
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      ports:
        - name: main-http
          port: 8000
          targetPort: 8000
      selector:
        app.kubernetes.io/component: app
        app.kubernetes.io/name: test-app
      type: ClusterIP
Check ALB annotation not set when when Owner missing:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        k: v
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      ports:
        - name: main-http
          port: 8000
          targetPort: 8000
      selector:
        app.kubernetes.io/component: app
        app.kubernetes.io/name: test-app
      type: ClusterIP
Check default port set:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: exporter
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      ports:
        - name: main-http
          port: 8000
          targetPort: 8000
      selector:
        app.kubernetes.io/component: exporter
        app.kubernetes.io/name: test-app
      type: ClusterIP
Check extraPorts:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: exporter
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      ports:
        - name: main-http
          port: 8000
          targetPort: 8000
        - name: main-p1
          port: 8081
          targetPort: 8081
        - name: main-p2
          port: 8082
          targetPort: 8082
        - name: custom-p3
          port: 8083
          targetPort: 8083
        - name: main-p4
          port: 8084
          targetPort: 8084
      selector:
        app.kubernetes.io/component: exporter
        app.kubernetes.io/name: test-app
      type: ClusterIP
Check name, namespace and labels:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: exporter
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      ports:
        - name: main-http
          port: 8000
          targetPort: 8000
      selector:
        app.kubernetes.io/component: exporter
        app.kubernetes.io/name: test-app
      type: ClusterIP
Check port and targetPort can be set from Service:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: exporter
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      ports:
        - name: main-http
          port: 22
          targetPort: 2222
      selector:
        app.kubernetes.io/component: exporter
        app.kubernetes.io/name: test-app
      type: ClusterIP
Check port can be overridden:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: exporter
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/application: test-app
        app.mintel.com/component: test-app
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      ports:
        - name: main-http
          port: 1234
          targetPort: 1234
      selector:
        app.kubernetes.io/component: exporter
        app.kubernetes.io/name: test-app
      type: ClusterIP
